SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1261,14 +1297,26 @@ void USpatialNetDriver::ServerReplicateActors_ProcessPrioritizedActors(UNetConne
 	int32 ActorUpdatesThisConnection = 0;
 	int32 ActorUpdatesThisConnectionSent = 0;
 
+	const int32 NumActorsMigrating = MigrationHandler.GetActorsToMigrate().Num();
+
 	// SpatialGDK - Entity creation rate limiting based on config value.
 	uint32 EntityCreationRateLimit = GetDefault<USpatialGDKSettings>()->EntityCreationRateLimit;
 	int32 MaxEntitiesToCreate = (EntityCreationRateLimit > 0) ? EntityCreationRateLimit : INT32_MAX;
+	if (MaxEntitiesToCreate < NumActorsMigrating)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("EntityCreationRateLimit of %i ignored because %i actors need to migrate"), EntityCreationRateLimit, NumActorsMigrating);
+		MaxEntitiesToCreate = NumActorsMigrating;
+	}
```

Comment:
Why do we need to update this if we're updating `MaxActorsToReplicate`?

Reply:
They could be different values ? Not sure I understand the question.

Reply:
Updating MaxEntitiesToCreate based on NumActorsMigrating seems ambiguous

Reply:
Sure. I'm setting it just in case this would actually prevent migration to happen.

Reply:
Looking at the code I don't think it should, and I think setting it here needlessly could believably troll someone in future

Reply:
Well I can see it trolling me.
@m-samiec  was of the opinion we should remove these anyway.

Reply:
Yeah this logic is pretty confusing, and I agree we should remove these values when we can. Unfortunately NWX still use the entity creation limiting. As we now guarantee that all migration actors should have the highest priority, could we just ignore these limits whilst we are still in that range?

Reply:
So right now the code would create at most Max(NumMigration, MaxNumEntities) . Are you suggesting I change it to create at most NumMigration + MaxNumEntities ?

Reply:
I'm saying we should leave in `MaxActorsToReplicate = NumActorsMigrating;` but remove `MaxEntitiesToCreate = NumActorsMigrating;`

Reply:
Latest modification to the code means that actors without entities would not be considered for migration anyway, so will remove.